modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaExtractSetPortletCheck.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Micaelle Silva
+ */
+public class UpgradeJavaExtractSetPortletCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> extendedClassNames = javaClass.getExtendedClassNames();
+
+		if (!extendedClassNames.contains("BasePanelApp")) {
+			return content;
+		}
+
+		Matcher matcher = _setPortletPattern.matcher(content);
+
+		if (matcher.find()) {
+			content = StringUtil.replace(
+				content, matcher.group(), _getNewImplementation(matcher));
+		}
+
+		return content;
+	}
+
+	private String _getNewImplementation(Matcher matcher) {
+		StringBundler sb = new StringBundler(18);
+
+		sb.append(matcher.group(1));
```

Comment:
@MicaelleSilvaa, we could delete this method and use all inside one string. With a `concat` direct on the `StringUtil.replace(`

Reply:
resolved